meshtastic/interfaces/ble/interface.py

```
@@ -2036,6 +1785,235 @@ def _sanitize_address(self, address: str | None) -> str | None:
+            discover_devices
+        ):
+            return cast(list[BLEDevice], discover_devices(address))
+        legacy_discover_devices = getattr(discovery_manager, "_discover_devices", None)
+        if callable(legacy_discover_devices) and not _is_unconfigured_mock_callable(
+            legacy_discover_devices
+        ):
+            return cast(list[BLEDevice], legacy_discover_devices(address))
+        # Preserve pre-refactor failure mode for misconfigured doubles.
+        fallback_discover_devices = discovery_manager.discover_devices
+        if _is_unconfigured_mock_callable(fallback_discover_devices):
+            raise AttributeError("Discovery manager is missing discover_devices")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)